CRM/Financial/BAO/Payment.php

```diff
@@ -486,6 +486,9 @@ protected static function getPayableLineItems($params): array {
     if ($outstandingBalance !== 0.0) {
       $ratio = $params['total_amount'] / $outstandingBalance;
     }
+    elseif ($params['total_amount'] < 0) {
+      $ratio = $params['total_amount'] / (float) CRM_Core_BAO_FinancialTrxn::getTotalPayments($params['contribution_id'], TRUE);
```

Comment:
@seamuslee001 what if CRM_Core_BAO_FinancialTrxn::getTotalPayments() = 0 or is less than 0

I suspect in the former case you would get a divided by zero error. 

I'm trying to get my head around the latter case but negative contributions are somewhat common whereas adding a refund to an already refunded contribution must be getting quite obscure but still possible I think

Reply:
@eileenmcnaughton I have now added in 2 unit tests one for adding an negative payment to an already refunded contribution and another adding another negative payment onto a negative contribution. In Neither case i encountered the fact that this was 0 because the $outstandingBalance was always non 0. Let me know what you think of the tests